### DISCUSSION TITLE
Open tree for current buffer

### DISCUSSION THREAD

**Discussion by harrisoncramer** [nvim-tree/nvim-tree.lua#3065] [Q&A] [answered]
First off, thank you for this awesome plugin! It's great.

Is it possible to configure the plugin so that the tree opens relative to the buffer's current location on your computer? For instance, I'll often originally open up nvim in `~/myfiles/folder/foo.js` and within the same session, edit a file in a different directory, for example at `~/anotherfolder/different/bar.js`. Currently, when inside that second directory, the tree opens within the original directory that nvim first opened in (the `~/myfiles/folder` directory).

Is it possible to open the tree dynamically based on which buffer is currently focused?

**Comment by msva** [comment, marked as answer]
`au BufEnter * if &ft != 'help' | silent! cd %:p:h | endif`

**Comment by msva** [comment]
(actually, checking for 'help' filetype is not that necessary, I guess)

**Comment by harrisoncramer** [comment]
Thank you! Worked perfectly, alongside the `vim_tree_respect_buf_cwd` setting.

**Comment by Toddgm** [comment]
> `au BufEnter * if &ft != 'help' | silent! cd %:p:h | endif`

sorry new to nvim, may i ask where should i add this command to, i have some problem as the issue owner has here.thx

**Comment by msva** [comment]
> where should i add this command to

[you should add it] to your neovim config.

Depends on what structure you use.

It can be either:
- `init.vim` (or some file `source`'ed from it) if you use vanilla vim-like configuration
- `init.lua` (or any file `require`'ed from it) if you prefer Lua.
- some files prepared for end-user customisations, if you use some novice-bundles like LazyVim and so on (don't know exact paths, ask in their community)
